app/logic/Calendar/Event.ts

```diff
@@ -192,11 +193,67 @@ export class Event extends Observable {
 
   async respondToInvitation(response: Responses): Promise<void> {
     assert(this.response > ResponseType.Organizer, "Only invitations can be responded to");
-    throw new AbstractFunction();
+    let accounts = appGlobal.emailAccounts.contents.filter(account => account.canSendInvitations && this.participants.some(participant => participant.response != ResponseType.Organizer && participant.emailAddress == account.emailAddress));
+    assert(accounts.length == 1, "Failed to find matching account for invitation");
+    let participant = this.participants.find(participant => participant.emailAddress == accounts[0].emailAddress);
+    participant.response = response;
+    await this.save();
+    await accounts[0].sendInvitationResponse(this, response);
   }
 
-  protected async sendInvitationResponse(response: Responses): Promise<void> {
-    throw new Error("Implement me!"); // TODO
+  /* Returns an icalEvent object suitable for NodeMailer */
+  getNMical(method?: iCalMethod): { method: iCalMethod, content: string } | null {
```

Comment:
Should be in ICalGenerator or similar, and accept `Event` as parameter

Reply:
Done in bf51300